src/debuginfo.cpp

```diff
@@ -811,14 +831,17 @@ void jl_getFunctionInfo(const char **name, size_t *line, const char **filename,
         if ((*it).second.func) {
             DISubprogram debugscope =
                 DISubprogram(prev.Loc.getScope((*it).second.func->getContext()));
-            *filename = debugscope.getFilename().data();
+            jl_copy_str(filename, debugscope.getFilename().data());
```

Comment:
Well, IIUC the assumption was that these did not need to be freed because the owning objects are live for the lifetime of the program (which  is probably a valid assumption until we e.g. start to GC the function objects?). Probably better to be explicit though.

> However, this is not a new assumptions with this PR.

Agreed. We seem to have gotten away with it so far.